include/deal.II/base/table_indices.h

```diff
@@ -81,294 +109,79 @@ class TableIndicesBase
- * class of a specializations.
- *
- * @ingroup data
- * @author Wolfgang Bangerth, Ralf Hartmann 2002
- */
-template <>
-class TableIndices<4> : public TableIndicesBase<4>
-{
-public:
-  /**
-   * Default constructor. Set all indices to zero.
-   */
-  TableIndices ();
-
-  /**
-   * Constructor. Set indices to the given values.
-   *
-   * The default values for the second and subsequent arguments are necessary
-   * for some neat template tricks in SymmetricTensor where we only want to
-   * set the first index and construct the subsequent ones later on, i.e. for
-   * the moment we don't care about the later indices.
-   */
-  TableIndices (const unsigned int index1,
-                const unsigned int index2 = numbers::invalid_unsigned_int,
-                const unsigned int index3 = numbers::invalid_unsigned_int,
-                const unsigned int index4 = numbers::invalid_unsigned_int);
-};
-
-
-/**
- * Array of indices of fixed size used for the TableBase class.
- *
- * This is the specialization for a five-dimensional table. This class only
- * differs in the non-default constructors from the other specializations.
- * Actual storage of and access to data is done by the TableIndicesBase base
- * class of a specializations.
- *
- * @ingroup data
- * @author Wolfgang Bangerth, Ralf Hartmann 2002
- */
-template <>
-class TableIndices<5> : public TableIndicesBase<5>
-{
-public:
-  /**
-   * Default constructor. Set all indices to zero.
-   */
-  TableIndices ();
-
-  /**
-   * Constructor. Set indices to the given values.
-   *
-   * The default values for the second and subsequent arguments are necessary
-   * for some neat template tricks in SymmetricTensor where we only want to
-   * set the first index and construct the subsequent ones later on, i.e. for
-   * the moment we don't care about the later indices.
-   */
-  TableIndices (const unsigned int index1,
-                const unsigned int index2 = numbers::invalid_unsigned_int,
-                const unsigned int index3 = numbers::invalid_unsigned_int,
-                const unsigned int index4 = numbers::invalid_unsigned_int,
-                const unsigned int index5 = numbers::invalid_unsigned_int);
-};
+/* --------------------- Template and inline functions ---------------- */
 
 
-/**
- * Array of indices of fixed size used for the TableBase class.
- *
- * This is the specialization for a six-dimensional table. This class only
- * differs in the non-default constructors from the other specializations.
- * Actual storage of and access to data is done by the TableIndicesBase base
- * class of a specializations.
- *
- * @ingroup data
- * @author Wolfgang Bangerth, Ralf Hartmann 2002
- */
-template <>
-class TableIndices<6> : public TableIndicesBase<6>
+template <int N>
+TableIndices<N>::TableIndices()
 {
-public:
-  /**
-   * Default constructor. Set all indices to zero.
-   */
-  TableIndices ();
+  Assert (N != 0, ExcMessage("Cannot create a TableIndices object of size 0"));
```

Comment:
I think you want to change this to `N>0` to also rule out negatives. In practice, I'm sure you get all sorts of compiler errors before you're even allowed to call this function (and get the run-time exception) so it may or may not matter whether we even have it.